src/workerd/api/queue.c++

```diff
@@ -453,7 +453,7 @@ jsg::Ref<QueueEvent> startQueueEvent(
                           jsg::alloc<QueueController>(event.addRef()),
                           jsg::JsValue(h.env.getHandle(js)).addRef(js),
                           h.getCtx());
-      event->waitUntil(kj::mv(promise));
+      event->waitUntil(promise.then([&event]{ event->setIsQueueHandlerComplete(); }));
```

Comment:
This isn't accounting for the possibility of the promise being rejected. We likely want to do the same thing in an error/catch handler, as in:

```
event->waitUntil(promise.then(
  [&event](){ event->setIsQueueHandlerComplete(); },
  [&event](kj::Exception&& e){ event->setIsQueueHandlerComplete(); }));
```

And at that point, we may as well also record whether it completed successfully or with an error, e.g. by adding a boolean `success` parameter or information about the exception type to that method.